### DISCUSSION TITLE
The auto-generated release notes do not include commits

### DISCUSSION THREAD

**Discussion by iilyak** [community/community#31628] [Pull Requests]
While implementing automatic release publishing workflow I planned to use `gh` tool to generate release. However I've noticed that the release notes generated in the release do not include any commits. The only content of the notes is 

```
<!-- Release notes generated using configuration in .github/release.yml at 2.23.0 -->

**Full Changelog**: https://github.com/iilyak/clouseau/compare/2.21.0...2.23.0
```

I've opened a ticket for `gh` tool https://github.com/cli/cli/issues/6159.

Further [investigation revealed](https://github.com/cli/cli/issues/6159#issuecomment-1232928805) that it seems to be a problem with github itself. 

When I attempt to create release via github UI I see a hint "There were no pull requests associated with the commits included in the release. Automatically-generated release notes were not generated". However there are merge commits in the repository. 

The commits in the repository https://github.com/iilyak/clouseau/commits/master

<img width="1649" alt="Screen Shot 2022-09-01 at 8 55 22 AM" src="https://user-images.githubusercontent.com/9804420/187959015-082fc90d-ef27-47b6-a4f9-e1b31042f045.png">

**Comment by nevernotmove** [comment]
Same problem and error message for me. I assumed the release notes would be generated and contain issues / commits. Did anyone find a way around it?

**Comment by codersasha** [comment]
I haven't found a way around it, but in case this helps -- this seems to occur for me when the commits are not on the "main" branch. 

In our org, we create cherry-pick tag chains for patch releases (i.e. we cut `v1.2.3` off `v1.2.2`, with cherry-picks from the main branch added onto it). When we try generate release notes for `v1.2.2...v1.2.3`, they are empty, even though the "Compare" link clearly shows the commits as in your screenshot.

**Reply by TDcaiqiang** [reply]
There is a similar case when releasing a `hotfix version` which is based on the previous release tag instead of the main branch, hotfix branch is not based on the main and should be based on the release version running online, because there may be commits that have been merged into the main after the latest release, but the hotfix does not want to include them and if based on main to release hotfix all of them need revert, it's not elegant.

**Reply by TDcaiqiang** [reply]
I suppose the reason is the hotfix tag which is created based on the previous release tag is not included in the main, so maybe merging the tag from the hotfix into the main could be a workaround, but it seems strange.

**Comment by MartinX3** [comment]
The sad problem is that they only respect changes from pull requests.
If you commit directly to your master branch it will be ignored.

See https://github.com/orgs/community/discussions/14517

**Reply by Ark2000** [reply]
lol

**Reply by sovetski** [reply]
very anoying, on the side projects where we are single developer who works on it, we generally work on the main branch, we can not spend time to create a pull request for each change to validate it by our self just after...

**Reply by mrdulin** [reply]
> very anoying, on the side projects where we are single developer who works on it, we generally work on the main branch, we can not spend time to create a pull request for each change to validate it by our self just after...

+1

**Reply by mrdulin** [reply]
They should put this on the [Automatically generated release notes](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes) documentation with **bold** text.  I wasted an hour testing pushing tags directly to the repository, but the automatically generated release notes didn’t work.

**Comment by Youssef11296** [comment]
any solution here?

**Comment by tomkeuper** [comment]
Is this something we can get fixed? I'm also running into the same problem. Im trying to create release notes on a production branch.
Maybe by adding an option to select what branch it looks at instead of just master?
I assume it looks at the default branch atleast

**Comment by GunnarMorrigan** [comment]
Obviously we would also like to see this changed...

Any updates from GitHub?

**Comment by ac185423** [comment]
My team is also experiencing this same issue. Release notes are not auto-generating when we make Github releases off of our main branch. There are new PR merge commits that should be included in the auto-generated release notes.

**Comment by ac185423** [comment]
Any update on this?

**Comment by ac185423** [comment]
Bumping the thread to keep it relevant. This is still an issue for my team. It really puts a kink in our release flow.

**Comment by fynnfluegge** [comment]
Issue is also relevant here.

I am fine that commits are not included but rather only references of merged PRs in release notes in general. 

But if I want to cherrypick merged PRs from main into a hotfix branch and create a release from that hotfix branch, even the cherrypicks of the merged PRs are not included in the autogenerated release notes. It would be really helpful if at least cherrypicks from merged PRs are recognised by the release notes generator. Currently this needs to be added manually.

**Comment by sovetski** [comment]
Any news please? On side projects we generally work on "main" branch without PR, which is logic, so it should include commits from main branch too...

**Comment by deadletterq** [comment]
Still relevant issue.

**Comment by vyamak** [comment]
Seems to me that only "Verified" commits are exported to release notes. Others are ignored. Why? I’m not sure. And so far, no way to change this rule, from my understanding.

**Reply by lexfernandez** [reply]
I just notice this is what is actually happening to me, those commits that are unverified are being ignored

**Reply by leavesster** [reply]
`Verified` commit that push directly to main is ignored too.

**Comment by miteshpc** [comment]
Creating a PR is the best practice but tideous and cumbersome. Would be possible for auto generated release notes to go down to the commits rather than the PR only, but then it would also generate lots of noise.

Maybe use of ChatGPT to convert the full changelog into release notes is more appropriate in this point of time.

**Comment by beatrizacbs** [comment]
This is still pretty relevant. I am facing this exat same issue. I think it would greatly improve developer experience.

**Comment by haydnjm** [comment]
Same issue here

**Reply by lexfernandez** [reply]
you will need to make your commits `Verified` and yours collaborators as well. otherwise github wont include then in the autogenerated notes

**Reply by lexfernandez** [reply]
here is a link to setup the signature and sign your commits.
https://docs.github.com/en/authentication/managing-commit-signature-verification

please note that existing commits wont be signed.

**Comment by vladshcherbin** [comment]
It's just bizarre such a simple thing does not exist in the cli 🤯

**Comment by SURAJ2600** [comment]
I am facing an issue related to this. Usually, after every release, generating the release notes includes all the PRs. However, since last week, when trying to create a release, all the old PR links have been added as part of the release.

I cross-checked the changes between develop and master, and the changes are valid. However, when comparing with the tags, it shows huge changes that were already merged into master a year ago.

I’ve confirmed that the tags we create are from the latest develop changes. Is anyone else facing this issue?

Additionally, when comparing the tags manually from the local machine, it shows the correct changes. But when generating the release notes or comparing the tag with the previous one, it shows invalid PR details

<img width="819" alt="Screenshot 2024-12-03 at 12 47 15 PM" src="https://github.com/user-attachments/assets/019e433b-e4cd-40e3-91b0-66b79ef50e71">

**Comment by khoadaniel** [comment]
Happened to us.

The release note is not consistent between the `What's Changed` and the `Full Changelog`.
This happened for the first time. The other releases notes did not have this behavior.


## What's Changed
![image](https://github.com/user-attachments/assets/29d130c4-4c44-4df1-b640-43db8b48d555)

## Full Changelog
![image](https://github.com/user-attachments/assets/fccc1cd7-ebbc-4178-8d00-20a1e10b9493)

**Comment by cf-tbasiuk** [comment]
Any updates on the issue? Seems to be broken still.

**Comment by kairusds** [comment]
still broken

**Comment by RohitKushvaha01** [comment]
as a workaround i use this

```bash
git log TAGNAME..HEAD --pretty="* %s by @%an"
```
